src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -20,58 +20,1244 @@
+        VERIFY_ALWAYS(
+            !( m_handler->accessTypeBackend == AccessType::READ_WRITE &&
+               ( !std::get< PE_NewlyCreated >( res_pair ) ||
+                 auxiliary::file_exists( fullPath(
+                     std::get< PE_InvalidatableFile >( res_pair ) ) ) ) ),
+            "Can only overwrite existing file in CREATE mode." );
+
+        if ( !std::get< PE_NewlyCreated >( res_pair ) )
+        {
+            auto file = std::get< PE_InvalidatableFile >( res_pair );
+            m_dirty.erase( file );
+            dropFileData( file );
+            file.invalidate( );
+        }
+
+        std::string const dir( m_handler->directory );
+        if ( !auxiliary::directory_exists( dir ) )
+        {
+            auto success = auxiliary::create_directories( dir );
+            VERIFY( success, "Could not create directory." );
+        }
+
+        associateWithFile( writable, shared_name );
+        this->m_dirty.emplace( shared_name );
+        getFileData( shared_name ).m_mode = adios2::Mode::Write; // WORKAROUND
+        // ↑ ADIOS2 does not yet implement ReadWrite Mode
+
+        writable->written = true;
+        writable->abstractFilePosition =
+            std::make_shared< ADIOS2FilePosition >( );
+    }
+}
+
+void ADIOS2IOHandlerImpl::createPath(
+    Writable * writable,
+    const Parameter< Operation::CREATE_PATH > & parameters )
+{
+    std::string path;
+    refreshFileFromParent( writable );
+
+    /* Sanitize path */
+    if ( !auxiliary::starts_with( parameters.path, '/' ) )
+    {
+        path = filePositionToString( setAndGetFilePosition( writable ) ) + "/" +
+            auxiliary::removeSlashes( parameters.path );
+    }
+    else
+    {
+        path = "/" + auxiliary::removeSlashes( parameters.path );
+    }
+
+    /* ADIOS has no concept for explicitly creating paths.
+     * They are implicitly created with the paths of variables/attributes. */
+
+    writable->written = true;
+    writable->abstractFilePosition = std::make_shared< ADIOS2FilePosition >(
+        path, ADIOS2FilePosition::GD::GROUP );
+
+    auto varName = nameOfVariable( writable );
+    // m_IO.DefineVariable< void >( varName );
+}
+
+void ADIOS2IOHandlerImpl::createDataset(
+    Writable * writable,
+    const Parameter< Operation::CREATE_DATASET > & parameters )
+{
+    if ( m_handler->accessTypeBackend == AccessType::READ_ONLY )
+    {
+        throw std::runtime_error( "Creating a dataset in a file opened as read "
+                                  "only is not possible." );
+    }
+    if ( !writable->written )
+    {
+        /* Sanitize name */
+        std::string name = auxiliary::removeSlashes( parameters.name );
+
+        auto file = refreshFileFromParent( writable );
+        auto filePos = setAndGetFilePosition( writable, name );
+        filePos->gd = ADIOS2FilePosition::GD::DATASET;
+        auto varName = filePositionToString( filePos );
+        // we use a unique_ptr to circumvent the fact that std::optional
```

Comment:
add same `@todo as above?`

Reply:
Btw, I you really want it for internal use, I can add optional-lite:
https://gist.githubusercontent.com/ax3l/b4ae0d439c8503cfc08b9c04d65f1327/raw/f750ada9a6460f4df05657827f70dea574165c9c/awesome_standalone.md

Reply:
I think this would be helpful to have. At least, I missed having something similar to this several times now, and it's arguably better than a type that also forces us to do heap allocations.

Reply:
I can provide you that, documented in #563.

We must just avoid to use it in user-facing header files.